version: "3"

services:
  db:
    image: mysql:5.7
    platform: linux/amd64 # 로컬 환경에서만
    environment:
      MYSQL_DATABASE: Boss
      MYSQL_USER: boss
      MYSQL_PASSWORD: password
      MYSQL_ROOT_PASSWORD: password
    ports:
      - "3307:3306"
    networks:
      - network
    volumes:
      - ./Backend/init.sql:/docker-entrypoint-initdb.d/init.sql
      - ./Backend/my.cnf:/etc/mysql/my.cnf
    healthcheck:
      test: ["CMD-SHELL", "mysqladmin ping -h localhost -u root -p password"]
      timeout: 5s
      retries: 10

  web: &web
    build:
      context: ./Backend
      dockerfile: Dockerfile
    ports:
      - "5001:5001"
    depends_on:
      - db
      - celery
    networks:
      - network
    restart: on-failure
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:5001/"]
      interval: 5s
      timeout: 3s
      retries: 3

  # ai:
  #   build:
  #     context: ./AI
  #     dockerfile: Dockerfile
  #   ports:
  #     - "5002:5002"
  #   depends_on:
  #     - db
  #   networks:
  #     - network
  #   restart: on-failure

  rabbitmq:
    image: rabbitmq
    ports:
      - "5672:5672"
    environment:
      - RABBITMQ_DEFAULT_USER=user
      - RABBITMQ_DEFAULT_PASS=password
    networks:
      - network

  celery:
    <<: *web
    command: celery -A celery_task.celery worker --loglevel=INFO
    depends_on:
      - rabbitmq
    ports: []

networks:
  network:
    driver: bridge
